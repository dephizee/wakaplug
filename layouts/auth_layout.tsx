// import styles from '../styles/AuthLayout.module.css'
import { GlobalStyle } from '../styles/globals.style';
import { AuthLayoutStyled } from '../components/auth_layout.style';
import { setEmail, setToken, setVerifcationCode } from '../redux/actions';
import { connect } from 'react-redux'
import { useEffect } from 'react';
import { useStateWithLocalStorage, useStateWithLocalStorageToken } from '../hooks/local_storage';

const mapStateToProps = (state:any) => {
    return {
        authData: state.authData
    }
}

const mapDispatchToProps = (dispatch:any) => {
    return {
        setStoreToken: (token:string) => dispatch(setToken(token)),
        setStoreEmail: (email:string) => dispatch(setEmail(email)),
        setStoreVerificationCode: (v_code:string) => dispatch(setVerifcationCode(v_code)),
    }
}

export const AuthMainLayout = ({children, setStoreEmail, setStoreToken, setStoreVerificationCode}:any) => {
    const [token,setLocalToken] = useStateWithLocalStorageToken();
    const [localEmail,setLocalEmail] = useStateWithLocalStorage('tmp_email');
    const [localVerificationCode,setVerificationCode] = useStateWithLocalStorage('tmp_verification_code');
    useEffect(() => {
        if(!token) return;
        setStoreToken(token)
    }, [setStoreToken, token]);
    useEffect(() => {
        if(!localEmail) return;
        setStoreEmail(localEmail)
    }, [setStoreEmail, localEmail]);
    useEffect(() => {
        if(!localVerificationCode) return;
        setStoreVerificationCode(localVerificationCode)
    }, [setStoreVerificationCode, localVerificationCode]);
    return ( 
        <AuthLayoutStyled>
            <GlobalStyle />
            {children}
        </AuthLayoutStyled>
     );
}

export const AuthLayout = connect(mapStateToProps, mapDispatchToProps)(AuthMainLayout)
