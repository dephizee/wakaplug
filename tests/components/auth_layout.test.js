import React from 'react';
import renderer from 'react-test-renderer';
import { AuthLayoutStyled} from "./../../components/auth_layout.style";


test('AuthLayoutStyled working', () => {
  const component = renderer.create(
    <AuthLayoutStyled>Bolt</AuthLayoutStyled>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});