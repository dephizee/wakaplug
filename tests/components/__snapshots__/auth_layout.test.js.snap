// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayoutStyled working 1`] = `
<div
  className="sc-bdvvtL dbwtDt"
>
  Bolt
</div>
`;
